### PR TITLE
Updated reference architectures to scale Mattermost

### PR DESCRIPTION
**Pull request by cwarnermm**
Outstanding work:
- expand available reference architectures section to explain what user count means (Neil B)
- populate Azure & GCP instance data for each scale page (Neil B & team)
- identify best place to link out to lifecycle/compatibility docs pages (Neil B)

### CODE REVIEW

**Comment by cwarnermm** [comment]
@agarciamontoro @isacikgoz @agnivade - Are you open to helping with the outstanding action items identified in the PR summary, please? Specifically, populating Azure & GCP instance data for each scale page, expanding on the load testing section, and suggesting any additional considerations for scaling up to 100 users?

**Review by nab-77 — CHANGES_REQUESTED** [review]
@cwarnermm first round of updates. This is a great layout. Thank you so much!

File: source/scale/scaling-for-enterprise.rst [hunk 1]
```diff
@@ -6,85 +6,38 @@ Scaling for Enterprise
...
+- `Up to 79000 users </scale/scale-to-79000-users.rst>`__
+- `Up to 88000 users </scale/scale-to-88000-users.rst>`__
 
-- See `hardware and software requirements for hardware sizing </install/software-hardware-requirements.html>`__.
+Load Testing
+------------
 
-Cluster-based deployment
-------------------------
+[Neil] Lay out what we did in testing these archs (specifically which version of MM was used in the tests, when we plan to publish updated lists etc).
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/scale/scaling-for-enterprise.rst:34]: @nab-77 - I've provided editorial feedback on this section. Please review to ensure that your original intent is clearly communicated.

File: source/scale/scaling-for-enterprise.rst [hunk 2]
```diff
@@ -6,85 +6,40 @@ Scaling for Enterprise
...
+Tests were defined by configuration of the actions executed by each simulated user (and the frequency of these actions). The coordinator metrics define a health system under load. Full testing methodology, configuration, and setup is available. Sign up at `Mattermost Community <https://community.mattermost.com/>`__ and join the `Developers: Performance channel <https://community.mattermost.com/core/channels/developers-performance>`__.
 
-To scale to 100,000 users and above, we recommended using `the Mattermost open source load testing framework <https://github.com/mattermost/mattermost-load-test-ng>`__ to simulate usage of your system at full scale.
+Mattermost load testing tools
+-----------------------------
 
-.. csv-table::
-    :header: "Feature", "Self-Hosted", "Amazon AWS", "Azure"
+Mattermost provides a set of tools written in Go to help profiling Mattermost under heavy load, simulating real-world usage of a server installation at scale. The `Mattermost Load Test Tools <https://github.com/mattermost/mattermost-load-test-ng>`__ estimates the maximum number of concurrently active users the target system supports, and enables you to control the load to generate.
```
Comment (agarciamontoro) [inline source/scale/scaling-for-enterprise.rst:43]: We usually refer to the tool in singular, but 2/5.

File: source/scale/scaling-for-enterprise.rst [hunk 3]
```diff
@@ -6,85 +6,40 @@ Scaling for Enterprise
...
-    "Load balancing", "Yes", "`Elastic Load Balancer <https://aws.amazon.com/elasticloadbalancing/>`__", "`Azure Load Balancer <https://azure.microsoft.com/en-us/services/load-balancer/>`__"
-    "Multi-region, HA storage", "Yes", "`S3 <https://aws.amazon.com/s3/>`__", "?"
-    "Multi-region, HA MySQL", "Yes - using a solution like Galera", "`Aurora <https://aws.amazon.com/rds/aurora/>`__ / `RDS for MySQL <https://aws.amazon.com/rds/mysql/>`__", "`Azure MySQL <https://azure.microsoft.com/en-us/services/mysql/>`__"
-    "Multi-region, HA PostgreSQL", "Yes", "`Aurora <https://aws.amazon.com/rds/aurora/>`__ / `RDS for PostgreSQL <https://aws.amazon.com/rds/postgresql/>`__", "`Azure PostgreSQL <https://azure.microsoft.com/en-us/services/postgresql/>`__"
-    "Multi-region, HA elasticsearch", "Yes", "`Amazon Elasticsearch Service <https://aws.amazon.com/opensearch-service/>`__", "No"
+Visit the `Mattermost Load Test Tools <https://github.com/mattermost/mattermost-load-test-ng/tree/master/docs>`__ documentation on GitHub for details on getting started with the tools, and visit `GDoc <https://godoc.org/github.com/mattermost/mattermost-load-test-ng>`__ for code-specific documentation.
```
Comment (agarciamontoro) [inline source/scale/scaling-for-enterprise.rst:45]: `GDoc` reads like Google Docs to me. I also usually link to https://pkg.go.dev/github.com/mattermost/mattermost-load-test-ng directly, not godoc.

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thank you for this, @cwarnermm and @nab-77! I left some comments inline about specific things, but here are some more general comments:

#### Notes on the diagram
- The first Mobile App Experience block says "Chrome Native app" instead of "Android native app"
- I'm not sure what the Authentication client, Authentication provider and Data Management Service entries mean in the Mattermost Server block.

#### Azure / GCP
We have not tested anything on Azure nor GCP. My recommendation is to remove those altogether. If we **need** to have them listed, I can look for similarly-speced instances in those services, but we'll have to add a big disclaimer stating that those are not tested.

#### Testing methodology and updates
I'm not sure how much information we want there, and how specific, but I think we could add some more things that can be interesting:
- No ElasticSearch was used
- No job servers were used
- All tests with more than one app node had an nginx running in front of them 
- Some more info on the DB:
  - We used Postgres v14
  - The initial database looked like:
    - Posts count: ~100 million
    - Users count: ~3000
    - Teams count: 20
    - Channels count: ~720000

File: source/scale/estimated-storage-per-user-per-month.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+ - Primarily using text messages and links to communicate. Examples would include software development teams that heavily use web-based document creation and management tools, and therefore rarely upload files to the server.
+
+- **Medium usage teams** (5-25 MB/user/month)
+
+ - Using a mix of text messages as well as shared documents and images to communicate. Examples might include business teams that may commonly drag and drop screenshots, PDFs and Microsoft Office documents into Mattermost for sharing and review.
+
+- **High usage teams** (25-100 MB/user/month)
+
+ - Heaviest utilization comes from teams uploading a high number of large files into Mattermost on a regular basis. Examples include creative teams who share and store artwork and media with tags and commentary in a pipeline production process.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/scale/estimated-storage-per-user-per-month.rst:18]: @nab-77 -- Thoughts?

File: source/scale/scale-to-1000-users.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** | **Azure Instance** | **GCP Instance** |
++========================+==========+==================+================+==================+====================+==================+
+| Mattermost Application | 2        | 4                | 1              | c6i.large        |                    |                  |
++------------------------+----------+------------------+----------------+------------------+--------------------+------------------+
+| RDS Writer             | 2        | 16               | 1              | db.r6g.large     |                    |                  |
++------------------------+----------+------------------+----------------+------------------+--------------------+------------------+
+| RDS Reader             | 16       | 16               | 1              | db.r6g.large     |                    |                  |
++------------------------+----------+------------------+----------------+------------------+--------------------+------------------+
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/scale/scale-to-1000-users.rst:26]: FYI @nab-77 ^

**Comment by cwarnermm** [comment]
@nab-77 - Regarding @agarciamontoro's feedback: "We have not tested anything on Azure nor GCP. My recommendation is to remove those altogether. If we need to have them listed, I can look for similarly-speced instances in those services, but we'll have to add a big disclaimer stating that those are not tested." -- how do we want to proceed via docs? Remove these columns?

File: source/scale/scale-to-70000-users.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
++------------------------+----------+------------------+----------------+------------------+
+| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
++========================+==========+==================+================+==================+
+| Mattermost Application | 16       | 32               | 4              | c6i.xlarge       |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Writer             | 16       | 128              | 1              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Reader             | 16       | 128              | 4              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
```
Comment (agarciamontoro) [inline source/scale/scale-to-70000-users.rst:26]:
```suggestion
+------------------------+----------+------------------+----------------+------------------+
| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
+========================+==========+==================+================+==================+
| Mattermost Application | 16       | 32               | 4              | c6i.4xlarge       |
+------------------------+----------+------------------+----------------+------------------+
| RDS Writer             | 16       | 128              | 1              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
| RDS Reader             | 16       | 128              | 4              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
```

File: source/scale/scale-to-79000-users.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
++------------------------+----------+------------------+----------------+------------------+
+| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
++========================+==========+==================+================+==================+
+| Mattermost Application | 16       | 32               | 5              | c6i.xlarge       |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Writer             | 16       | 128              | 1              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Reader             | 16       | 128              | 4              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
```
Comment (agarciamontoro) [inline source/scale/scale-to-79000-users.rst:26]:
```suggestion
+------------------------+----------+------------------+----------------+------------------+
| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
+========================+==========+==================+================+==================+
| Mattermost Application | 16       | 32               | 5              | c6i.4xlarge       |
+------------------------+----------+------------------+----------------+------------------+
| RDS Writer             | 16       | 128              | 1              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
| RDS Reader             | 16       | 128              | 4              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
```

File: source/scale/scale-to-88000-users.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
++------------------------+----------+------------------+----------------+------------------+
+| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
++========================+==========+==================+================+==================+
+| Mattermost Application | 16       | 32               | 5              | c6i.xlarge       |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Writer             | 16       | 128              | 1              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
+| RDS Reader             | 16       | 128              | 4              | db.r6g.xlarge    |
++------------------------+----------+------------------+----------------+------------------+
```
Comment (agarciamontoro) [inline source/scale/scale-to-88000-users.rst:26]:
```suggestion
+------------------------+----------+------------------+----------------+------------------+
| **Resource Type**      | **vCPU** | **Memory (GiB)** | **# of Nodes** | **AWS Instance** |
+========================+==========+==================+================+==================+
| Mattermost Application | 16       | 32               | 6              | c6i.4xlarge       |
+------------------------+----------+------------------+----------------+------------------+
| RDS Writer             | 16       | 128              | 1              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
| RDS Reader             | 16       | 128              | 4              | db.r6g.4xlarge    |
+------------------------+----------+------------------+----------------+------------------+
```

File: source/scale/scaling-for-enterprise.rst [hunk 9]
```diff
@@ -6,85 +6,40 @@ Scaling for Enterprise
...
-
-For more information, check out our `Administrator Onboarding Tasks </getting-started/admin-onboarding-tasks.html#administrator-tasks>`__ documentation.
-
-Hosting recommendation for 100,000+ users
------------------------------------------
+Tests were defined by configuration of the actions executed by each simulated user (and the frequency of these actions) where the coordinator metrics define a health system under load. Tests were performed using the Mattermost v8.1 extended support release (ESR). Elasticsearch and job servers weren't used. All tests wtih more than a single app node had an NGINX proxy running in front of them.
 
-The following matrix presents key features for a successful multi-region Mattermost implementation that scales to 100,000 users with support for high availability and geographically based traffic routing in on premises, AWS, and Azure deployments.
+Full testing methodology, configuration, and setup is available. Sign up at `Mattermost Community <https://community.mattermost.com/>`__ and join the `Developers: Performance channel <https://community.mattermost.com/core/channels/developers-performance>`__.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/scale/scaling-for-enterprise.rst:38]: I believe the intention is that users interested in accessing details on the full testing methodology need to join the Mattermost Community Server and join the Performance channel to do so. @nab-77 - Is my understanding correct?

**Review by agarciamontoro — APPROVED** [review]
I have to log off for the day, so approving now to unblock. I'll leave the unaddressed comment to you both :) Thanks!

File: source/scale/scaling-for-enterprise.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (nab-77) [reply source/scale/scaling-for-enterprise.rst]: Correct @cwarnermm

**Comment by nab-77** [comment]
Great first pass team. v2 will take this to the next level.

**Pull request merged**